### DISCUSSION TITLE
SET SESSION group_concat_max_len

### DISCUSSION THREAD

**Discussion by touteslatitudes** [sequelize/sequelize#15377] [Help & Questions] [answered]
I need to increase the default group_concat_max_len parameter for a JSON_ARRAYAGG query

With Sequelize 6.21 and Mariadb 10.5.18,  I try the following query but this doesn't work with Sequelize.  The query is correct and runs with PhpMyAdmin and Mariadb.

var query = "SET SESSION group_concat_max_len = 10485760;  SELECT JSON_OBJECT(...) FROM my_table WHERE ....."
db.sequelize.query(query, { type: db.sequelize.QueryTypes.SELECT })

How can i change the group_concat_max_len with Sequelize for the session ?

Thanks for your answers

**Comment by ephys** [comment, marked as answer]
mysql, mysql2, and mariadb (the npm libraries) don't let you execute more than one statement per query (as an extra security layer against sql injection), you need to enable the `multipleStatements` option:

```ts
const sequelize = new Sequelize({
  // ...your other options

  // options passed as-is to mysql.createConnection
  dialectOptions: {
    multipleStatements: true,
  },
});
```

I haven't tested it but according to https://stackoverflow.com/questions/23266854/node-mysql-multiple-statements-in-one-query, as well as https://github.com/mariadb-corporation/mariadb-connector-nodejs/blob/7e8696f8926ee5793bc9c73413c572f957af7e9b/documentation/connection-options.md, it should work

**Reply by touteslatitudes** [reply]
Thanks Zoé. Tested and correct.

I have added : 
dialectOptions: {
    multipleStatements: true,
  },

With 2 statements (SET SESSION and SELECT) the result of the Select is in data[1] not data[0].

SET SESSION group_concat_max_len = 10485760; is mandatory for a big  JSON_ARRAYAGG aggregation result.
